examples/cifar10/cifar10-sim/README.md

```diff
@@ -0,0 +1,178 @@
+```
+source ./virtualenv/set_env.sh
+```
+install required packages for training
+```
+pip install --upgrade pip
+pip install -r ./virtualenv/min-requirements.txt
+```
+(optional) if you would like to plot the TensorBoard event files as shown below, please also install
+```
+pip install -r ./virtualenv/plot-requirements.txt
+```
+Set `PYTHONPATH` to include custom files of this example:
+```
+export PYTHONPATH=${PWD}/..
+```
+
+### 2. Download the CIFAR-10 dataset 
+To speed up the following experiments, first download the [CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) dataset:
+```
+python3 ../pt/utils/cifar10_download_data.py
+```
+
+## 3. Run simulated FL experiments
+
+We are using NVFlare's [simulator]() to run the following experiments. 
+
+First set the output root where to save the results
+```
+export RESULT_ROOT=/tmp/nvflare/sim_cifar10
+```
+
+### 3.1 Varying data heterogeneity of data splits
+
+We use an implementation to generated heterogeneous data splits from CIFAR-10 based a Dirichlet sampling strategy 
+from FedMA (https://github.com/IBM/FedMA), where `alpha` controls the amount of heterogeneity, 
+see [Wang et al.](https://arxiv.org/abs/2002.06440).
+
+We use `set_alpha.sh` to change the alpha value inside the job configurations.
+
+### 3.2 Centralized training
+
+To simulate a centralized training baseline, we run FL with 1 client for 25 local epochs but only for one round. 
+It takes circa 6 minutes on an NVIDIA TitanX GPU.
+```
+./set_alpha.sh cifar10_central 0.0
+nvflare simulator job_configs/cifar10_central --workspace ${RESULT_ROOT}/central --threads 1 --n_clients 1
+```
+Note, here `alpha=0.0` means that no heterogeneous data splits are being generated.
+
+You can visualize the training progress by running `tensorboard --logdir=${RESULT_ROOT}`
+![Central training curve](./figs/central_training.png)
+
+### 3.3 FedAvg on different data splits
+
+FedAvg (8 clients). Here we run for 50 rounds, with 4 local epochs. Corresponding roughly 
+to the same number of iterations across clients as in the central baseline above (50*4 divided by 8 clients is 25):
+Each job will take about 35 minutes, depending on your system. 
+
+You can copy the whole block into the terminal, and it will execute each experiment one after the other.
+```
+./set_alpha.sh cifar10_fedavg 1.0
+nvflare simulator job_configs/cifar10_fedavg --workspace ${RESULT_ROOT}/fedavg_alpha1.0 --threads 8 --n_clients 8
+./set_alpha.sh cifar10_fedavg 0.5
+nvflare simulator job_configs/cifar10_fedavg --workspace ${RESULT_ROOT}/fedavg_alpha0.5 --threads 8 --n_clients 8
+./set_alpha.sh cifar10_fedavg 0.3
+nvflare simulator job_configs/cifar10_fedavg --workspace ${RESULT_ROOT}/fedavg_alpha0.3 --threads 8 --n_clients 8
+./set_alpha.sh cifar10_fedavg 0.1
+nvflare simulator job_configs/cifar10_fedavg --workspace ${RESULT_ROOT}/fedavg_alpha0.1 --threads 8 --n_clients 8
+```
+
+### 3.4: Advanced FL algorithms (FedProx, FedOpt, and SCAFFOLD)
```

Comment:
should do `3.4 ` instead of `3.4:` to be consistent

Reply:
Thanks. Changed.